test/parallel_api/numeric/numeric.ops/reduce_by_segment_non_primitive_types.pass.cpp

```diff
@@ -0,0 +1,56 @@
+// -*- C++ -*-
```

Comment:
I think we don't need to add a new test file. We already have reduce_by_segment.pass.cpp.
All that we need  - to add new C++ type into the existed test. We have such approach in the most of OneDPL tests.

Reply:
Fixed: expanded reduce_by_segment test; new test was removed.